tools/bootstrap2pb.cc

```diff
@@ -31,8 +32,9 @@ int main(int argc, char** argv) {
   Envoy::PlatformImpl platform_impl_;
   Envoy::Stats::IsolatedStoreImpl stats_store;
   Envoy::Event::RealTimeSystem time_system; // NO_CHECK_FORMAT(real_time)
+  Envoy::Random::RandomGeneratorImpl rand;
   Envoy::Api::Impl api(platform_impl_.threadFactory(), stats_store, time_system,
-                       platform_impl_.fileSystem());
+                       platform_impl_.fileSystem(), rand);
```

Comment:
cc @akonradi @lizan @htuch 

I wonder if it would be possible to have a basic test the exercises this tool and catches compile breakages like this one.

Reply:
The compilation breakage should have been caught by the existing build rule. Do we not try to build everything under //tools somewhere in CI?

Reply:
We must not which seems broken. Can you add this somewhere here? https://github.com/envoyproxy/envoy/blob/master/ci/do_ci.sh

/wait

Reply:
My understanding is that bazel test does not build binaries that are not required to run tests.  In order to verify compilation of a binary, you need a test that depends on it.

Reply:
bazel build //tools/... should suffice.

Reply:
Yeah ^

Reply:
I think that the way to do this would be to change the following line in ci/build_setup.sh to also cover //tools/...
I can offer to send that as a separate change if we prefer just getting this build fix in and address the CI coverage issue separately.

[ -z "${ENVOY_BUILD_TARGET}" ] && export ENVOY_BUILD_TARGET=//source/exe:envoy-static //tools/...

Reply:
There are other build breakages under //tools/... like:

ERROR: tools/testdata/protoxform/envoy/v2/BUILD:23:14: //tools/testdata/protoxform/envoy/v2:freeze_protos: missing input file '//tools/testdata/protoxform/envoy/v2:active_non_terminal.proto'
ERROR: tools/testdata/protoxform/envoy/v2/BUILD:23:14 3 input file(s) do not exist

I'm looking into them.